### DISCUSSION TITLE
How to use Engine.query method?

### DISCUSSION THREAD

**Discussion by m5261136** [UPPAALModelChecker/UPPAAL-Meta#215] [Q&A] [answered]
I followed the documentation for class Engine and tried to verify using the query method, but a syntax error was output.

qr = engine.query(system, "search-order 0\n"
					+ "S 1\n"
					+ "state-representation 0\n"
					+ "diagnostic 0\n"
					+ "extrapolation 0\n"
					+ "hashtable-size 27\n"
					+ "reuse-statespace 1\n"
					+ "parametric-comp 1\n"
					+ "modest 0\n"
					+ "statistical 0.01 0.01 0.05 0.05 0.05 0.9 1.1 0.0 0.0 1280.0 0.01", query, qf);

Is the writing of the second argument of the query method correct?

**Comment by thorulf4** [comment, marked as answer]
It is currently not possible to specify engine options in the query method. You would have to use the `engine.setOptionSettings` method and then call query.

Here is an example of how to do this:
```java
EngineSettings options = engine.getOptions().get().getDefaultSettings();
options.addSetting("-S", "1");
options.addSetting("--state-representation", "0");
options.addSetting("--diagnostic", "0");
options.addSetting("--extrapolation", "0");
// Add remaining options here

engine.setOptionSettings(options);
engine.query(system, query, qf);
```
